### PR TITLE
ffmpeg: add option --enable-nvdec

### PR DESCRIPTION
**Pull request by dataCobra**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64 (crossbuild)
  - aarch64-musl (crossbuild)
  - armv7l (crossbuild)

This is an update to try fixing issue #38782.

@nvidiaLinuxUser, could you please test this PR for your system?

### CODE REVIEW

File: srcpkgs/ffmpeg/template [hunk 1]
```diff
@@ -26,19 +26,20 @@ makedepends="zlib-devel bzip2-devel freetype-devel alsa-lib-devel libXfixes-deve
  $(vopt_if webp libwebp-devel) $(vopt_if sofa libmysofa-devel)
  $(vopt_if vulkan 'vulkan-loader Vulkan-Headers')
  $(vopt_if drm libdrm-devel) $(vopt_if svtav1 libsvt-av1-devel)
- $(vopt_if srt srt-devel) $(vopt_if rist librist-devel)"
+ $(vopt_if srt srt-devel) $(vopt_if rist librist-devel)
+ $(vopt_if nvdec nv-codec-headers)"
```
Comment (sgn) [inline srcpkgs/ffmpeg/template:30]: Please add `configure_args` option to enable/disable nvenc/nvdec on build_options, let's say someone want to enable nvdec and disable nvenc for whatever reason.

Comment (sgn) [reply srcpkgs/ffmpeg/template:30]: It would be wise to put nvdec and nvenc near each other

Comment (dataCobra) [reply srcpkgs/ffmpeg/template]: Hey @sgn,

I've made the changes as requested.
I've also refactored `makedepends=` so the entries are in the correct text width.

**Comment by dataCobra** [comment]
@nvidiaLinuxUser could you give feedback here as soon as you tested the changes?

**Comment by sgn** [comment]
Unless you properly figure out what wrong with your system, we can't help you.

**Comment by abenson** [comment]
Working well for me.

**Comment by dataCobra** [comment]
Hey @abenson,

> Working well for me.

Did something not work for you before testing this PR?

If so, can you please explain what your problem was and what you used?

**Comment by abenson** [comment]
> Did something not work for you before testing this PR?

No, I just mean with this PR I was able to force decoding on the nVidia GPU.

**Comment by dataCobra** [comment]
@sgn, for me this is also working well. Maybe we could just merge the change? It doesn't have
any down sights as far as I can tell.

**Comment by sgn** [comment]
Maybe ask @abenson  He uses NVIDIA, I don't see much value myself, I don't do NVIDIA, the other guy can't fix his problem :-p

**Pull request merged**